### PR TITLE
Allow unversioned project to select a specific version of a versioned library

### LINKED ISSUE #323: Allow specifying a versioned dependency from an unversioned spec
If I don't care to version my API but I want to leverage a library that has versions, I should still be able to use the versioned library. Today, you have to provide a mapping from your versions the library versions. Probably we can just allow specifying the library version without the mapping in the `@versionedDependency` decorator.

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix #323

in library: 
```cadl
@versioned("1" | "2")
namespace VersionedLib {
  model Foo {
    name: string;
    @added("2") age: int32;
  }
}
```

in project use version 1 of library
```cadl
@versionedDependency(VersionedLib, "1")
namespace MyService;

model Test extends VersionedLib.Foo {}
```

### CODE REVIEW

File: packages/openapi3/src/openapi.ts [hunk 1]
```diff
@@ -247,18 +247,14 @@ function createOAPIEmitter(program: Program, options: OpenAPIEmitterOptions) {
     }
     const versions = getVersionRecords(program, serviceNs);
     for (const record of versions) {
-      record.projections.push({
```
Comment (timotheeguerin) [inline packages/openapi3/src/openapi.ts:250]: @bterlson was this meant to be here. It causes some issue with the change below with `enableProjections` always running

Comment (markcowl) [reply packages/openapi3/src/openapi.ts:250]: I believe that the toTypescript projection is currently the default projection that all emitters must run.  The design issue on side cars (#218) calls for clarifying a standard set of projections and which are supposed to run.

File: packages/versioning/test/test-versioned-dependencies.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        @versioned("v1" | "v2")
+        @versionedDependency(VersionedLib, "1")
+        namespace MyService {
+          model Test extends VersionedLib.Foo {}
+        } 
+    `);
+      expectDiagnostics(diagnostics, {
+        code: "@cadl-lang/versioning/versioned-dependency-record-not-model",
+        message:
+          "The versionedDependency decorator must provide a model mapping local versions to dependency 'dependency' versions",
```
Comment (markcowl) [inline packages/versioning/test/test-versioned-dependencies.ts:107]: nice

**Review by markcowl — APPROVED** [review]
I think it would be best to get Brian's feedback about the toTypescript projection before merging, but LGTM.

File: packages/openapi3/src/openapi.ts (hunk 1 above)
Comment (bterlson) [reply packages/openapi3/src/openapi.ts:250]: Fine with removing this. It somewhat helpful for demos but it was mistakenly adding it so removing it is fine.

**Review by bterlson — APPROVED** [review]
Excellent!

**Pull request merged**